### ISSUE TITLE
OCI: The specified item already exists in the keychain, Still

### ISSUE THREAD

**Issue by jeremymcgee73** [hashicorp/terraform-provider-helm#989]
### Terraform, Provider, Kubernetes and Helm Versions

```
Terraform version: 1.1.7
Provider version: 3.7.1
Kubernetes version: 1.21
```

### Affected Resource(s)

- helm_release

### Terraform Configuration Files
```hcl
data "aws_ecr_authorization_token" "token" {}

locals {
  helm_base_registry_path = trimprefix(data.aws_ecr_authorization_token.token.proxy_endpoint, "https://")
}

resource "helm_release" "example1" {
  name    = "example1"
  chart   = "oci://${local.helm_base_registry_path}/example1"
  version = "1.0.0"

  repository_username = data.aws_ecr_authorization_token.token.user_name
  repository_password = data.aws_ecr_authorization_token.token.password
}

resource "helm_release" "example2" {
  name    = "example2"
  chart   = "oci://${local.helm_base_registry_path}/example2"
  version = "2.0.0"

  repository_username = data.aws_ecr_authorization_token.token.user_name
  repository_password = data.aws_ecr_authorization_token.token.password
}
```

### Debug Output

```
Error: could not login to OCI registry "XXXXXXXX.dkr.ecr.us-east-1.amazonaws.com": error storing credentials - err: exit status 1, out: `The specified item already exists in the keychain.`
```

### Steps to Reproduce

1. `terraform apply`

### References

- #845

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jeremymcgee73** [comment]
I'm still running into this, I believe a fix would be to check if the credential already exists. We could also create a new credential based on the resource name instead. This happens when we run multiple helm releases in parallel, with OCI.

**Comment by jeremymcgee73** [comment]
This may also be OSX related only. I should be able to let you know if thats the case soon..

**Comment by jeremymcgee73** [comment]
Also, this may only be a problem on OSX.

**Comment by jeremymcgee73** [comment]
Just thought this would be helpful, tested with `2.8.0` and still had the same problem on OSX.

**Comment by jeremymcgee73** [comment]
This only happens on OSX BTW. I have not seen it on Centos/AL2.

**Comment by MikiLoz92** [comment]
Happening here as well, after upgrading TF from 1.2 to 1.3.8 & the provider version (now on 2.9.0). Mac OS X Ventura 13.1. 
I'm creating 4 concurrent instances of the same chart.

**Comment by LudoNieuwenhuizen** [comment]
My colleagues on Dell do not have this issue, I do, with my Macbook. 

This is quite a blocker for me, as we install ~8 helm_releases in the same directory for most of our projects, and I now have to target each resource individually.
I tried a suggestion fro another ticket I saw, to make the helm_releases dependent on each other. I added a `depends_on` on each of my `helm_release` blocks and made each one reference the previous one so that there's a real chaining dependency tree, with the first one not having any dependencies. This still resulted in:
```[bash]
error storing credentials - err: exit status 1, out: `The specified item already exists in the keychain.`
```
Interestingly, when I `terraform apply` multiple times (without `-target`), I do not get the same erroring helm_release back all the time. Sometimes I get the error for 4 helm_projects, and the next time it can be 3 different ones. 
Can it be that eventhough the helm_release depend on each other, they sort of install simultaneous?

**Comment by armandjanssen1978** [comment]
Same here, just moved to a Macbook and running into this (for me) blocking issue.

**Comment by grzegorzlyczba** [comment]
@jeremymcgee73 it's an issue with `docker-credential-osxkeychain` which is used as a helper to store/get items from osx keychain.
Can you post what version your'e running (docker-credential-osxkeychain version)?
As a workaround I can suggest you to skip the keychain store, you find the settings in the file `~/Library/Preferences/helm/registry/config.json` (you can remove `"credsStore": "osxkeychain"` entry).

**Comment by LudoNieuwenhuizen** [comment]
Hi there, 

For me, this is 
$ docker-credential-osxkeychain version
0.6.4

Where should I update that config? I call the helm_release provider using the terraform binary, which is the only thing I run locally

**Comment by grzegorzlyczba** [comment]
> Hi there, 
> 
> 
> 
> For me, this is 
> 
> $ docker-credential-osxkeychain version
> 
> 0.6.4
> 
> 
> 
> Where should I update that config? I call the helm_release provider using the terraform binary, which is the only thing I run locally

The file is on your computer

**Comment by ricbra** [comment]
@grzegorzlyczba Thanks for your suggestion, you just saved my day 👍🏻

**Comment by LudoNieuwenhuizen** [comment]
This is really great, thanks a bunch @grzegorzlyczba! Works very well

**Comment by yebolenko** [comment]
Thank you a lot. Suggestion to remove osxkeychain just saved me a day

**Comment by jeremymcgee73** [comment]
Thank you! I was just running into this again, and this solved my problem.

**Comment by rjinski** [comment]
To add to this, it also happens when trying to specific the registry on two different helm provider aliases.

```hcl
provider "helm" {
  registry {
    url      = "oci://${data.aws_caller_identity.current.account_id}.dkr.ecr.${data.aws_region.current.name}.amazonaws.com"
    username = data.aws_ecr_authorization_token.token.user_name
    password = data.aws_ecr_authorization_token.token.password
  }
  experiments {
    manifest = true
  }
}

provider "helm" {
  alias = "no_manifest"
  registry {
     url      = "oci://${data.aws_caller_identity.current.account_id}.dkr.ecr.${data.aws_region.current.name}.amazonaws.com"
     username = data.aws_ecr_authorization_token.token.user_name
     password = data.aws_ecr_authorization_token.token.password
   }
  experiments {
    manifest = false
  }
}
```

**Comment by ascopes** [comment]
any update on this? I'm getting this almost every time I run helm today via Terraform.

**Comment by ascopes** [comment]
I will try and see if I can raise a PR for this...

**Comment by ascopes** [comment]
I've raised the above PR to work around this error, it should be able to work around the issue until such a time that osxkeychain is fixed or the underlying libraries handle this edge case correctly.

**Comment by zoispag** [comment]
@ascopes workaround PR looks promising. Is there any chance it can be reviewed, merged, released?

It is impossible to run `terraform apply` without specifying the `-target`.

**Comment by acherifi** [comment]
Still got the same error on my side using multiple helm providers with aliases. Any solutions ? I can only rely on CI right now to apply my terraform which is not ideal

**Comment by julian3xl** [comment]
if @grzegorzlyczba says so, it should be fixed!
